modules/Movecraft/src/main/java/net/countercraft/movecraft/craft/ICraft.java

```diff
@@ -96,56 +96,66 @@ public void rotate(Rotation rotation, MovecraftLocation originPoint) {
                 getNotificationPlayer().sendMessage(I18nSupport.getInternationalisedString("Rotation - Turning Too Quickly"));
             return;
         }
+        setLastRotateTime(System.nanoTime());
         if (getType().getUseGravity()) {
-            MoveOnRotate move = moveUp(rotation, originPoint);
+            final MoveOnRotate move = moveUp(rotation, originPoint);
             final Craft craft = this;
-            if (move.equals(MoveOnRotate.UP)) {
-                translate(0, 1, 0);
-                new BukkitRunnable(){
-                    @Override
-                    public void run() {
-                        Movecraft.getInstance().getAsyncManager().submitTask(new RotationTask(craft, originPoint, rotation, craft.getW()), craft);
-                    }
-                }.runTaskLaterAsynchronously(Movecraft.getInstance(), 4);
-                return;
-            } else if (move.equals(MoveOnRotate.DOWN)) {
-                Movecraft.getInstance().getAsyncManager().submitTask(new RotationTask(craft, originPoint, rotation, craft.getW()), craft);
-                new BukkitRunnable(){
-                    @Override
-                    public void run() {
-                        translate(0, -1, 0);
-                    }
-                }.runTaskLaterAsynchronously(Movecraft.getInstance(), 4);
-                return;
+            switch (move){
+                case UP:
+                    translate(0, 1, 0);
+                    break;
+                case DOWN:
+                    Movecraft.getInstance().getAsyncManager().submitTask(new RotationTask(craft, originPoint, rotation, craft.getW()), craft);
+                    break;
             }
+            new BukkitRunnable(){
+                @Override
+                public void run() {
+                    switch (move){
+                        case DOWN:
+                            translate(0, -1, 0);
+                            break;
+                        case UP:
+                            Movecraft.getInstance().getAsyncManager().submitTask(new RotationTask(craft, originPoint, rotation, craft.getW()), craft);
```

Comment:
This will not move the craft up

Reply:
It is being moved up before the BukkitRunnable is run